README.md

```diff
@@ -323,7 +309,7 @@ declare module '@ember/controller' {
 
 You'll need to add that module and interface declaration to all your existing service and controller declarations for this to work (again, see the [blog post][pt4] for further details), but once you do that, you'll have this much nicer experience throughout! It's not quite vanilla Ember.js, but it's close—and this way, you still get all those type-checking and auto-completion benefits, but with a lot less noise! Moreover, you actually get a significant benefit over "vanilla" Ember: we type-check that you typed the key correctly in the `service` invocation.
 
-If you have a reason to fall back to just getting the `Service` or `Controller` types, you can always do so by just using the string-less variant: `service('session')` will check that the string is a valid name of a service; `session()` will not.
+> If you get the compile error `Assertion Failed: InjectedProperties should be defined with the inject computed property macros.` then you're using the old `inject` helper on at least one ES6 class in Ember 3.x. Replace it with the `@service` decorator instead as above.
```

Comment:
> If you have a reason to fall back to just getting the `Service` or `Controller` types, you can always do so by just using the string-less variant: `service('session')` will check that the string is a valid name of a service; `session()` will not.

I don't think this para is correct any more. What should we write for those who wish to use ember-cli-typescript without ember-decorators? I'm nowhere near expert enough (I'm a newbie to TS and ES6) and I'm writing this because I spent three times as long getting it to compile as writing the app in the first place.

Reply:
something like this should be the recommended pattern for non-decorator users

```ts
import { inject as service } from '@ember/service';
import Component from '@ember/component';

class MyComponent extends Component.extend({
   store: service('store')
}) {
   // normal member functions go here (they'll be auto-placed on the prototype)
   // normal member data goes here (they'll be auto-placed on each instance)
}

```